qiskit_experiments/framework/composite/composite_experiment.py

```diff
@@ -85,26 +82,34 @@ def _set_backend(self, backend):
         for subexp in self._experiments:
             subexp._set_backend(backend)
 
-    def _add_job_metadata(self, experiment_data, jobs, **run_options):
-        # Add composite metadata
-        super()._add_job_metadata(experiment_data, jobs, **run_options)
+    def _additional_metadata(self):
+        return {"component_job_metadata": []}
```

Comment:
This is intended to store experiment level metadata which for this experiment is the list of the experiment level metadata for each component experiment. This is so they can be added to each component experiment metadata during analysis in case those component experiments analysis requires it (which I think curve fit based ones do). Previously this wasn't needed because the component containers were all initialized with their metadata during run.

I don't really want to add changing how base experiment class metadata is handled in this PR since it will further complicate things, but this is something to revisit in a follow up for sure.